tpls/desul/include/desul/atomics/Common.hpp

```diff
@@ -83,11 +83,11 @@ struct numeric_limits_max;
 
 template <>
 struct numeric_limits_max<uint32_t> {
-  static constexpr uint32_t value = 0xffffffffu;
+  static constexpr uint32_t value = -1;
```

Comment:
An alternative expression that makes the 'all ones' nature totally clear, and avoids pedantic details about types and numeric behavior, would be
```suggestion
  static constexpr uint32_t value = ~static_cast<uint32_t>(0);
```

Reply:
I like this.

Reply:
This being Phil's suggestion.

Reply:
Discussion needs to happen on the Desul side and until proof of the contrary no warnings are being emitted with the code I proposed and modulo arithmetic is as easy as it gets so there is no need to try to be clever.

Reply:
yeah its fine leave the -1